include/seqan3/core/algorithm/configuration_utility.hpp

```diff
@@ -0,0 +1,153 @@
+ *
+ * \see seqan3::detail::is_configuration_valid_v
+ */
+template <config_element_concept query_t, config_element_concept ... compare_types>
+struct is_configuration_valid
+{
+    //!\privatesection
+    //!\brief Helper typedef.
+    using algorithm_id_type = remove_cvref_t<decltype(query_t::id)>;
+    //!\brief Helper typedef.
+    using index_type        = std::underlying_type_t<algorithm_id_type>;
+
+    //!\publicsection
+    //!\brief The result of checking the compatibility.
+    static constexpr bool value =
+        (std::is_same_v<algorithm_id_type, remove_cvref_t<decltype(compare_types::id)>> && ...) &&
+        (compatibility_table<algorithm_id_type>[static_cast<index_type>(query_t::id)]
+                                             [static_cast<index_type>(compare_types::id)] && ...);
+};
+
+/*!\brief Helper variable template to check for valid configuration compositions.
+ * \ingroup algorithm
+ * \see seqan3::detail::is_configuration_valid
+ */
+template <typename query_t, typename ... compare_types>
+inline constexpr bool is_configuration_valid_v = is_configuration_valid<query_t, compare_types...>::value;
+
+// ----------------------------------------------------------------------------
+// Metafunction is_same_configuration_f
+// ----------------------------------------------------------------------------
+
+/*!\brief Helper meta function to check if a template type is contained in a seqan3::configuration.
+ * \ingroup algorithm
+ *
+ * \details
+ *
+ * This helper meta function is used to provide the `get` and `value_or` interface for template template types.
+ */
+template <template <typename ...> typename query_t>
+struct is_same_configuration_f
+{
+    /*!\brief A type template that evaluates to std::true_type if the given type is a specialization of `query_t`,
+     *        otherwise std::false_type.
+     * \tparam compare_type The type to compare against `query_t`.
+     */
+    template <typename compare_type>
+    using invoke = is_type_specialisation_of<compare_type, query_t>;
+};
+
+// ----------------------------------------------------------------------------
+// Metafunction is_algorithm_configuration
+// ----------------------------------------------------------------------------
+
+/*!\brief Value metafunction that returns whether a type is an algorithm configuration.
+ * \ingroup algorithm
+ *
+ * \returns std::true_type if the given type is a seqan3::detail::configuration, else std::false_type.
+ */
+template <typename object_t>
+struct is_algorithm_configuration : std::false_type
+{};
+
+//!\cond
+template <typename ...config_elements_t>
+struct is_algorithm_configuration<seqan3::configuration<config_elements_t...>> : std::true_type
+{};
+//!\endcond
+
+/*!\brief Helper variable template for seqan3::detail::is_algorithm_configuration.
+ * \ingroup algorithm
+ */
+template <typename object_t>
+inline constexpr bool is_algorithm_configuration_v = is_algorithm_configuration<object_t>::value;
```

Comment:
here again, can't we just use is_type_specialisation_of wherever this would be used?